CMakeLists.txt

```diff
@@ -51,6 +51,20 @@ if(MSVC)
     xcheck_add_c_compiler_flag(-Wno-reserved-identifier)
     xcheck_add_c_compiler_flag(-Wdeprecated-declarations)
     xcheck_add_c_compiler_flag(/experimental:c11atomics)
+    xcheck_add_c_compiler_flag(/wd4018)
+    xcheck_add_c_compiler_flag(/wd4061)
+    xcheck_add_c_compiler_flag(/wd4100)
+    xcheck_add_c_compiler_flag(/wd4242)
+    xcheck_add_c_compiler_flag(/wd4244)
+    xcheck_add_c_compiler_flag(/wd4245)
+    xcheck_add_c_compiler_flag(/wd4267)
+    xcheck_add_c_compiler_flag(/wd4388)
+    xcheck_add_c_compiler_flag(/wd4389)
+    xcheck_add_c_compiler_flag(/wd4710)
+    xcheck_add_c_compiler_flag(/wd4711)
+    xcheck_add_c_compiler_flag(/wd4820)
+    xcheck_add_c_compiler_flag(/wd4996)
+    xcheck_add_c_compiler_flag(/wd5045)
```

Comment:
It would be helpful to have comments explaining what warnings they suppress and/or mentions their gcc counterpart, e.g.
```
xcheck_add_c_compiler_flag(/wd4100) # -Wno-unused-parameter
```